api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -22,6 +22,7 @@
 import java.util.List;
 import java.util.Map;
 import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.RowKey;
```

Comment:
Do we need to introduce a createTable method that could accept `RowKey` in this interface ?  I rise this question because when I  support [flink SQL primary key](https://github.com/openinx/incubator-iceberg/pull/4/files#diff-45ca81857734430272d8132ed4008d1ce97560168a6d2c2d660295cbf265f4f9R57-R77) , I find it's necessary method.  Of course,  we could publish separate  PR for this if possible.

Reply:
Opening a separate issue for this is good enough for me now.

Reply:
That's a good point, I thought about that, and there were 2 reasons that let me decide to not add it:

1. currently both Spark and Hive SQL specification does not allow a primary key clause. I would expect the user to run something like `ALTER TABLE ... ADD ROW KEY` (actual syntax TBD) to add this row key in Spark.

2. it's now becoming hard to iterate through all the combinations of parameters in `createTable`, we do not have the methods with SortOrder in parameter yet, although it can be mapped to `SORTED BY` clause. I would prefer us to switch to use the table builder if possible instead  of adding all those overloading methods.

Reply:
Okay,  I agree it's good to use `TableBuilder` to `createTable` in future usage, as we are introducing more and more arguments when creating table.